src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -32,14 +32,27 @@ public partial class JsonTypeInfo
         // Add method delegate for non-generic Stack and Queue; and types that derive from them.
         internal object? AddMethodDelegate { get; set; }
 
-        internal JsonPropertyInfo? DataExtensionProperty { get; private set; }
+        internal JsonPropertyInfo? DataExtensionProperty { get; set; }
 
         // If enumerable or dictionary, the JsonTypeInfo for the element type.
         private JsonTypeInfo? _elementTypeInfo;
 
         // Avoids having to perform an expensive cast to JsonTypeInfo<T> to check if there is a Serialize method.
         internal bool HasSerialize { get; set; }
 
+        // Configure would normally have thrown why initializing properties for source gen but type had SerializeHandler
+        // so it is allowed to be used for serialization but it will throw if used for deserialization
+        internal bool ThrowOnDeserialize { get; set; }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Better let JIT make that call? It's likely that it would be inlined in any case.

Reply:
this has two references and one of them is in the hot (fast) path, I think it's better to inline this